libvuln/updates.go

```diff
@@ -0,0 +1,79 @@
+package libvuln
+
+import (
+	"context"
+	"io"
+	"net/http"
+
+	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/rs/zerolog"
+
+	"github.com/quay/claircore/internal/updater"
+	"github.com/quay/claircore/libvuln/driver"
+)
+
+func NewUpdater(pool *pgxpool.Pool, client *http.Client, config map[string]driver.ConfigUnmarshaler, workers int, filter func(string) bool) (*UpdateDriver, error) {
+	return &UpdateDriver{
+		exe: &updater.Online{
+			Pool:    pool,
+			Workers: workers,
+			Filter:  filter,
+		},
+		client: client,
+		config: config,
+	}, nil
+}
+
+func NewOfflineUpdater(config map[string]driver.ConfigUnmarshaler, filter func(string) bool, out io.Writer) (*UpdateDriver, error) {
+	return &UpdateDriver{
+		exe: &updater.Offline{
+			Filter: filter,
+			Output: out,
+		},
+		config: config,
+	}, nil
+}
+
+type UpdateDriver struct {
+	exe    updater.Controller
+	client *http.Client
+	config map[string]driver.ConfigUnmarshaler
+}
+
+func (d *UpdateDriver) RunUpdaters(ctx context.Context, fs ...driver.UpdaterSetFactory) error {
```

Comment:
libvuln.loopUpdaters and this method feel like they belong closer together. Consider the design of this separation and if its cohesive to have the summation of the looping functionality split between objects.